pkg/planner/core/planbuilder.go

```diff
@@ -4217,8 +4217,22 @@ func (b *PlanBuilder) buildImportInto(ctx context.Context, ld *ast.ImportIntoStm
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.FilePriv, "", "", "", plannererrors.ErrSpecificAccessDenied.GenWithStackByArgs("FILE"))
 	}
 	tableInfo := p.Table.TableInfo
-	tableInPlan, ok := b.is.TableByID(tableInfo.ID)
+	// we use the latest IS to support IMPORT INTO dst FROM SELECT * FROM src AS OF TIMESTAMP '2020-01-01 00:00:00'
+	// Note: we need to get p.Table when preprocessing, at that time, IS of session
```

Comment:
My understanding is, `p.Table.TableInfo` may be a latest or stale TableInfo. And in expectation the target table of IMPORT INTO should always be latest TableInfo.

And my suggestions are:
1. Can we modify the preprocessing to add a special branch for IMPORT INTO? I see there's such a special branch for `InsertStmt` https://github.com/pingcap/tidb/blob/134d2bfbe0002b24177155873038188d4544e538/pkg/planner/core/preprocess.go#L249-L253
2. This PR uses a "may be stale" table ID with a "latest" IS. This is a little weird. Can we use `TableByName` to get the latest TableInfo again and check the latest table ID is the same as `tableInfo.ID`?

Reply:
i don't want to support the case that user uses import into when the session is already in stale-read state by set `tidb_snapshot`

> Can we modify the preprocessing to add a special branch for IMPORT INTO?

that makes no difference, both requires special handling

> Can we use TableByName to get the latest TableInfo again and check the latest table ID is the same as tableInfo.ID?

checking existence of `maybe stale id`  VS  get latest id and compare, both lgtm, as tidb doesn't reuse id.

Reply:
>> Can we modify the preprocessing to add a special branch for IMPORT INTO?
>
>that makes no difference, both requires special handling

The difference is we get a latest TableInfo vs we get an uncertain TableInfo and check it has the same table ID with latest one. Will there be problems that the TableInfo of two different versions have same table ID? For example, some DDL has modified the columns or indices and table ID is not changed.

Reply:
when will this happen without in a stale-read session?

Reply:
Seems only happen in the session that already set `tidb_snapshot`. In this function we report error only when latest TableInfo is not found, so I'm afraid an old version TableInfo with the same table ID will cause some problems, when it's accessed after this function and before `handleNoDelayExecutor`. I didn't see problems now, so it's OK.

And in this page https://docs.pingcap.com/tidb/stable/basic-features#advanced-sql-features I found 2 terminology
"Stale read" and "Read historical data", I prefer these comment use "history snapshot read", "tidb_snapshot read" or some thing rather than "stale read" when refer to  `tidb_snapshot`

Reply:
we're mixing them in here https://docs.pingcap.com/tidb/stable/as-of-timestamp